code/lib/preview-api/src/modules/preview-web/render/StoryRender.ts

```diff
@@ -180,33 +180,16 @@ export class StoryRender<TRenderer extends Renderer> implements Render<TRenderer
     const abortSignal = (this.abortController as AbortController).signal;
 
     try {
-      let loadedContext: Awaited<ReturnType<typeof applyLoaders>>;
-      await this.runPhase(abortSignal, 'loading', async () => {
-        loadedContext = await applyLoaders({
-          ...this.storyContext(),
-          viewMode: this.viewMode,
-          // TODO add this to CSF
-          canvasElement,
-        } as unknown as StoryContextForLoaders<TRenderer>);
-      });
-      if (abortSignal.aborted) return;
-
-      const renderStoryContext: StoryContext<TRenderer> = {
-        ...loadedContext!,
-        // By this stage, it is possible that new args/globals have been received for this story
-        // and we need to ensure we render it with the new values
-        ...this.storyContext(),
```

Comment:
Before `this.storyContext()` would get called twice in one render. Once before loaders where run and once just before passing it down to `renderToCanvas`. This is not necessary, as when args/globals change, we will rerender the story with the new values.

This is also reflected in this test case:
https://github.com/storybookjs/storybook/blob/054f661b124965acdce09a5a03d1f375b153fdc3/code/lib/preview-api/src/modules/preview-web/PreviewWeb.test.ts#L912-L985

Reply:
cc @tmeasday and @JReinhold this is the recent that this test changed. And it seems to me like the right expected behavior. 

Ideally, we would cancel the loading and before each phase I would think in a rerender. I could do that in a subsequent PR. But I'm not sure about the reasons why we don't do this yet.

Reply:
@tmeasday How would a `loader` or` beforeEach` trigger a `setArg` call? It can not use the `useArgs` hook. Only the `decorators` and `render` function can call hooks.

Loaders ande beforeEach could do: 
`context.args.x = Math.random()` 

And this wouldn't trigger a rerender or infinite loop.

Reply:
Oh, good point. I was thinking of the use case of a decorator setting an arg with `useArgs`, but that would come after.